In [1]:
import pandas as pd

# Read CSV files using Pandas

In [2]:
df=pd.read_csv("../data/train.csv")



## Number of rows in the given datastore

In [3]:
len(df)

93686

## display first rows

In [4]:
df.head(10)

category  \
0      Online and Social Media Related Crime   
1                     Online Financial Fraud   
2                   Online Gambling  Betting   
3      Online and Social Media Related Crime   
4                     Online Financial Fraud   
5                     Online Financial Fraud   
6                     Online Financial Fraud   
7                     Online Financial Fraud   
8  RapeGang Rape RGRSexually Abusive Content   
9                      Any Other Cyber Crime   

                        sub_category  \
0  Cyber Bullying  Stalking  Sexting   
1                  Fraud CallVishing   
2           Online Gambling  Betting   
3                   Online Job Fraud   
4                  Fraud CallVishing   
5                 UPI Related Frauds   
6                  Fraud CallVishing   
7     Internet Banking Related Fraud   
8                                NaN   
9                              Other   

                                  crimeaditionalinfo  
0  I had continue received random calls and abusi...  
1  The above fraudster is continuously messaging ...  
2  He is acting like a police and demanding for m...  
3  In apna Job I have applied for job interview f...  
4  I received a call from lady stating that she w...  
5  FRAUD \t UPI PAYTM \r\nBANK \tPunjab National ...  
6  Sir I am Prabhat Singh jat  An app on playstor...  
7                               FINANCIAL FRAUD  RS   
8  I got the message on Whatsapp to my number The...  
9  Details entered in pdf file Person posing as A...

## display columns names

In [5]:
df.columns


Index(['category', 'sub_category', 'crimeaditionalinfo'], dtype='object')

## display dataset information

In [6]:
df.describe()


category        sub_category  \
count                    93686               87095   
unique                      15                  35   
top     Online Financial Fraud  UPI Related Frauds   
freq                     57434               26856   

                                       crimeaditionalinfo  
count                                               93665  
unique                                              85013  
top     Respected Sir\r\n\r\nA very serious matter I w...  
freq                                                 2342

## Number of NaN values for particular columns

In [7]:
df.isnull().sum()

category                 0
sub_category          6591
crimeaditionalinfo      21
dtype: int64

## We have to handle NaN values in the given dataset
- here we can see that there are 21 records that have no crimeaditionalinfo so we can drop that rows

In [8]:
df_cleaned = df.dropna(subset=['crimeaditionalinfo'])


In [9]:
df_cleaned.isnull().sum()

category                 0
sub_category          6591
crimeaditionalinfo       0
dtype: int64

### Process Sub Category
- Here we can see that there are 6591 NaN values in the sub category so we can drop that also or we have to fill that by replacing some values.

In [10]:
df_cleaned.shape

(93665, 3)

In [11]:
## New Df
df=df_cleaned

In [12]:
nan_sub_category = df[df['sub_category'].isna()]
nan_sub_category.head(15)

category sub_category  \
8    RapeGang Rape RGRSexually Abusive Content          NaN   
25   RapeGang Rape RGRSexually Abusive Content          NaN   
39                       Sexually Explicit Act          NaN   
45                   Sexually Obscene material          NaN   
49                       Sexually Explicit Act          NaN   
72                       Sexually Explicit Act          NaN   
77   RapeGang Rape RGRSexually Abusive Content          NaN   
79                       Sexually Explicit Act          NaN   
92   RapeGang Rape RGRSexually Abusive Content          NaN   
94                   Sexually Obscene material          NaN   
98   RapeGang Rape RGRSexually Abusive Content          NaN   
117  RapeGang Rape RGRSexually Abusive Content          NaN   
122                  Sexually Obscene material          NaN   
125                      Sexually Explicit Act          NaN   
129  RapeGang Rape RGRSexually Abusive Content          NaN   

                                    crimeaditionalinfo  
8    I got the message on Whatsapp to my number The...  
25   Respected Sir\r\n\r\nA very serious matter I w...  
39   httpswwwxnxxtvvideousapbfuckkkarrr\r\n\r\n Abo...  
45   Many fake accounts are created and Im sufferin...  
49   SirMaam \r\nThis is my third report on this re...  
72   This girl name is geeta  Iss ladki neh normal ...  
77   Respected Sir\r\n\r\nA very serious matter I w...  
79   Someone has created a profile on app called ok...  
92   Respected Sir\r\n\r\nA very serious matter I w...  
94   In twitter we have seen illegally publishing a...  
98   Respected Sir\r\n\r\nA very serious matter I w...  
117  Respected Sir\r\n\r\nA very serious matter I w...  
122  Someone post my nude on Instagram and when I a...  
125  The particular user is making sexual videos ab...  
129  Respected Sir\r\n\r\nA very serious matter I w...

In [13]:
df

category  \
0      Online and Social Media Related Crime   
1                     Online Financial Fraud   
2                   Online Gambling  Betting   
3      Online and Social Media Related Crime   
4                     Online Financial Fraud   
...                                      ...   
93681                 Online Financial Fraud   
93682                 Online Financial Fraud   
93683                 Online Financial Fraud   
93684  Online and Social Media Related Crime   
93685                 Online Financial Fraud   

                            sub_category  \
0      Cyber Bullying  Stalking  Sexting   
1                      Fraud CallVishing   
2               Online Gambling  Betting   
3                       Online Job Fraud   
4                      Fraud CallVishing   
...                                  ...   
93681     Internet Banking Related Fraud   
93682              EWallet Related Fraud   
93683                 UPI Related Frauds   
93684           Online Matrimonial Fraud   
93685                  Fraud CallVishing   

                                      crimeaditionalinfo  
0      I had continue received random calls and abusi...  
1      The above fraudster is continuously messaging ...  
2      He is acting like a police and demanding for m...  
3      In apna Job I have applied for job interview f...  
4      I received a call from lady stating that she w...  
...                                                  ...  
93681  Identity theft   Smishing SMS Fraud  CreditDeb...  
93682  RECEIVED CALL FROM  NUMBER ASKING ABOUT phone ...  
93683  Cyber Stalking   Blackmailing   PhoneSMSVOIP C...  
93684  Call karke bola ki aapka lotary laga ha aru AC...  
93685  There is app name koko loan app they send the ...  

[93665 rows x 3 columns]

In [14]:
nan_sub_category = df[df['sub_category'].isna()]
nan_sub_category

category sub_category  \
8      RapeGang Rape RGRSexually Abusive Content          NaN   
25     RapeGang Rape RGRSexually Abusive Content          NaN   
39                         Sexually Explicit Act          NaN   
45                     Sexually Obscene material          NaN   
49                         Sexually Explicit Act          NaN   
...                                          ...          ...   
93632                      Sexually Explicit Act          NaN   
93648                      Sexually Explicit Act          NaN   
93653  RapeGang Rape RGRSexually Abusive Content          NaN   
93667  RapeGang Rape RGRSexually Abusive Content          NaN   
93678  RapeGang Rape RGRSexually Abusive Content          NaN   

                                      crimeaditionalinfo  
8      I got the message on Whatsapp to my number The...  
25     Respected Sir\r\n\r\nA very serious matter I w...  
39     httpswwwxnxxtvvideousapbfuckkkarrr\r\n\r\n Abo...  
45     Many fake accounts are created and Im sufferin...  
49     SirMaam \r\nThis is my third report on this re...  
...                                                  ...  
93632    ob cash                                     ...  
93648  I got fraud by atm exchange in union bank of i...  
93653  Respected Sir\r\n\r\nA very serious matter I w...  
93667  Respected Sir\r\n\r\nA very serious matter I w...  
93678  Respected Sir\r\n\r\nA very serious matter I w...  

[6591 rows x 3 columns]

In [15]:
nan_sub_category.isnull().sum()

category                 0
sub_category          6591
crimeaditionalinfo       0
dtype: int64

## Replace NaN Sub Category with Appropriate value


In [16]:
unique_categories = df[df['sub_category'].isna()]['category'].unique()
unique_categories=unique_categories.tolist()

In [17]:
unique_categories

['RapeGang Rape RGRSexually Abusive Content',
 'Sexually Explicit Act',
 'Sexually Obscene material',
 'Child Pornography CPChild Sexual Abuse Material CSAM']

In [18]:
# Iterate through each unique category
for category in unique_categories:
    # Replace NaN in sub_category with the category name
    df.loc[(df['sub_category'].isna()) & (df['category'] == category), 'sub_category'] = category
    
    # Add a new category label for Women/Child Related Crime
    df.loc[(df['sub_category'] == category), 'category'] = 'Women/Child Related Crime'


In [19]:
df.isnull().sum()

category              0
sub_category          0
crimeaditionalinfo    0
dtype: int64

### So now there is not any NaN values in our dataset

In [20]:
df['category'].unique()

array(['Online and Social Media Related Crime', 'Online Financial Fraud',
       'Online Gambling  Betting', 'Women/Child Related Crime',
       'Any Other Cyber Crime', 'Cyber Attack/ Dependent Crimes',
       'Cryptocurrency Crime',
       'Hacking  Damage to computercomputer system etc',
       'Cyber Terrorism', 'Online Cyber Trafficking', 'Ransomware',
       'Report Unlawful Content'], dtype=object)

#### As of now we can keep these all categories and no need to process this

In [21]:
df['sub_category'].unique()

array(['Cyber Bullying  Stalking  Sexting', 'Fraud CallVishing',
       'Online Gambling  Betting', 'Online Job Fraud',
       'UPI Related Frauds', 'Internet Banking Related Fraud',
       'RapeGang Rape RGRSexually Abusive Content', 'Other',
       'Profile Hacking Identity Theft',
       'DebitCredit Card FraudSim Swap Fraud', 'EWallet Related Fraud',
       'Data Breach/Theft', 'Cheating by Impersonation',
       'Denial of Service (DoS)/Distributed Denial of Service (DDOS) attacks',
       'FakeImpersonating Profile', 'Cryptocurrency Fraud',
       'Sexually Explicit Act', 'Sexually Obscene material',
       'Malware Attack', 'Business Email CompromiseEmail Takeover',
       'Email Hacking', 'Hacking/Defacement',
       'Unauthorised AccessData Breach', 'SQL Injection',
       'Provocative Speech for unlawful acts', 'Ransomware Attack',
       'Cyber Terrorism',
       'Child Pornography CPChild Sexual Abuse Material CSAM',
       'Tampering with computer source documents',
      

In [22]:
type(df)

pandas.core.frame.DataFrame

In [23]:
df['sub_category']=df['sub_category'].replace('Cyber Bullying  Stalking  Sexting','Cyber Bullying/Stalking/Sexting')
df['sub_category']=df['sub_category'].str.strip()
df['sub_category'].unique()

array(['Cyber Bullying/Stalking/Sexting', 'Fraud CallVishing',
       'Online Gambling  Betting', 'Online Job Fraud',
       'UPI Related Frauds', 'Internet Banking Related Fraud',
       'RapeGang Rape RGRSexually Abusive Content', 'Other',
       'Profile Hacking Identity Theft',
       'DebitCredit Card FraudSim Swap Fraud', 'EWallet Related Fraud',
       'Data Breach/Theft', 'Cheating by Impersonation',
       'Denial of Service (DoS)/Distributed Denial of Service (DDOS) attacks',
       'FakeImpersonating Profile', 'Cryptocurrency Fraud',
       'Sexually Explicit Act', 'Sexually Obscene material',
       'Malware Attack', 'Business Email CompromiseEmail Takeover',
       'Email Hacking', 'Hacking/Defacement',
       'Unauthorised AccessData Breach', 'SQL Injection',
       'Provocative Speech for unlawful acts', 'Ransomware Attack',
       'Cyber Terrorism',
       'Child Pornography CPChild Sexual Abuse Material CSAM',
       'Tampering with computer source documents',
       '

In [24]:
df['sub_category']=df['sub_category'].replace('Fraud CallVishing','Fraud Call/Vishing')
df['sub_category']=df['sub_category'].str.strip()
df['sub_category'].unique()

array(['Cyber Bullying/Stalking/Sexting', 'Fraud Call/Vishing',
       'Online Gambling  Betting', 'Online Job Fraud',
       'UPI Related Frauds', 'Internet Banking Related Fraud',
       'RapeGang Rape RGRSexually Abusive Content', 'Other',
       'Profile Hacking Identity Theft',
       'DebitCredit Card FraudSim Swap Fraud', 'EWallet Related Fraud',
       'Data Breach/Theft', 'Cheating by Impersonation',
       'Denial of Service (DoS)/Distributed Denial of Service (DDOS) attacks',
       'FakeImpersonating Profile', 'Cryptocurrency Fraud',
       'Sexually Explicit Act', 'Sexually Obscene material',
       'Malware Attack', 'Business Email CompromiseEmail Takeover',
       'Email Hacking', 'Hacking/Defacement',
       'Unauthorised AccessData Breach', 'SQL Injection',
       'Provocative Speech for unlawful acts', 'Ransomware Attack',
       'Cyber Terrorism',
       'Child Pornography CPChild Sexual Abuse Material CSAM',
       'Tampering with computer source documents',
       

In [25]:
df['sub_category']=df['sub_category'].replace('RapeGang Rape RGRSexually Abusive Content','Rape/Gang Rape-Sexually Abusive Content')
df['sub_category']=df['sub_category'].str.strip()
df['sub_category'].unique()

array(['Cyber Bullying/Stalking/Sexting', 'Fraud Call/Vishing',
       'Online Gambling  Betting', 'Online Job Fraud',
       'UPI Related Frauds', 'Internet Banking Related Fraud',
       'Rape/Gang Rape-Sexually Abusive Content', 'Other',
       'Profile Hacking Identity Theft',
       'DebitCredit Card FraudSim Swap Fraud', 'EWallet Related Fraud',
       'Data Breach/Theft', 'Cheating by Impersonation',
       'Denial of Service (DoS)/Distributed Denial of Service (DDOS) attacks',
       'FakeImpersonating Profile', 'Cryptocurrency Fraud',
       'Sexually Explicit Act', 'Sexually Obscene material',
       'Malware Attack', 'Business Email CompromiseEmail Takeover',
       'Email Hacking', 'Hacking/Defacement',
       'Unauthorised AccessData Breach', 'SQL Injection',
       'Provocative Speech for unlawful acts', 'Ransomware Attack',
       'Cyber Terrorism',
       'Child Pornography CPChild Sexual Abuse Material CSAM',
       'Tampering with computer source documents',
       'D

In [26]:
df['sub_category']=df['sub_category'].replace('DebitCredit Card FraudSim Swap Fraud','Debit/Credit Card Fraud Or SIM Swap Fraud')
df['sub_category']=df['sub_category'].str.strip()
df['sub_category'].unique()

array(['Cyber Bullying/Stalking/Sexting', 'Fraud Call/Vishing',
       'Online Gambling  Betting', 'Online Job Fraud',
       'UPI Related Frauds', 'Internet Banking Related Fraud',
       'Rape/Gang Rape-Sexually Abusive Content', 'Other',
       'Profile Hacking Identity Theft',
       'Debit/Credit Card Fraud Or SIM Swap Fraud',
       'EWallet Related Fraud', 'Data Breach/Theft',
       'Cheating by Impersonation',
       'Denial of Service (DoS)/Distributed Denial of Service (DDOS) attacks',
       'FakeImpersonating Profile', 'Cryptocurrency Fraud',
       'Sexually Explicit Act', 'Sexually Obscene material',
       'Malware Attack', 'Business Email CompromiseEmail Takeover',
       'Email Hacking', 'Hacking/Defacement',
       'Unauthorised AccessData Breach', 'SQL Injection',
       'Provocative Speech for unlawful acts', 'Ransomware Attack',
       'Cyber Terrorism',
       'Child Pornography CPChild Sexual Abuse Material CSAM',
       'Tampering with computer source documents

In [27]:
df['sub_category']=df['sub_category'].replace('Business Email CompromiseEmail Takeover','Business Email Compromise/Email Takeover')
df['sub_category']=df['sub_category'].str.strip()
df['sub_category'].unique()

array(['Cyber Bullying/Stalking/Sexting', 'Fraud Call/Vishing',
       'Online Gambling  Betting', 'Online Job Fraud',
       'UPI Related Frauds', 'Internet Banking Related Fraud',
       'Rape/Gang Rape-Sexually Abusive Content', 'Other',
       'Profile Hacking Identity Theft',
       'Debit/Credit Card Fraud Or SIM Swap Fraud',
       'EWallet Related Fraud', 'Data Breach/Theft',
       'Cheating by Impersonation',
       'Denial of Service (DoS)/Distributed Denial of Service (DDOS) attacks',
       'FakeImpersonating Profile', 'Cryptocurrency Fraud',
       'Sexually Explicit Act', 'Sexually Obscene material',
       'Malware Attack', 'Business Email Compromise/Email Takeover',
       'Email Hacking', 'Hacking/Defacement',
       'Unauthorised AccessData Breach', 'SQL Injection',
       'Provocative Speech for unlawful acts', 'Ransomware Attack',
       'Cyber Terrorism',
       'Child Pornography CPChild Sexual Abuse Material CSAM',
       'Tampering with computer source document

In [28]:
df['sub_category']=df['sub_category'].replace('Unauthorised AccessData Breach','Unauthorised Access/Data Breach')
df['sub_category']=df['sub_category'].str.strip()
df['sub_category'].unique()

array(['Cyber Bullying/Stalking/Sexting', 'Fraud Call/Vishing',
       'Online Gambling  Betting', 'Online Job Fraud',
       'UPI Related Frauds', 'Internet Banking Related Fraud',
       'Rape/Gang Rape-Sexually Abusive Content', 'Other',
       'Profile Hacking Identity Theft',
       'Debit/Credit Card Fraud Or SIM Swap Fraud',
       'EWallet Related Fraud', 'Data Breach/Theft',
       'Cheating by Impersonation',
       'Denial of Service (DoS)/Distributed Denial of Service (DDOS) attacks',
       'FakeImpersonating Profile', 'Cryptocurrency Fraud',
       'Sexually Explicit Act', 'Sexually Obscene material',
       'Malware Attack', 'Business Email Compromise/Email Takeover',
       'Email Hacking', 'Hacking/Defacement',
       'Unauthorised Access/Data Breach', 'SQL Injection',
       'Provocative Speech for unlawful acts', 'Ransomware Attack',
       'Cyber Terrorism',
       'Child Pornography CPChild Sexual Abuse Material CSAM',
       'Tampering with computer source documen

In [29]:
df['sub_category']=df['sub_category'].replace('Child Pornography CPChild Sexual Abuse Material CSAM','Child Pornography/Child Sexual Abuse Material (CSAM)')
df['sub_category']=df['sub_category'].str.strip()
df['sub_category'].unique()

array(['Cyber Bullying/Stalking/Sexting', 'Fraud Call/Vishing',
       'Online Gambling  Betting', 'Online Job Fraud',
       'UPI Related Frauds', 'Internet Banking Related Fraud',
       'Rape/Gang Rape-Sexually Abusive Content', 'Other',
       'Profile Hacking Identity Theft',
       'Debit/Credit Card Fraud Or SIM Swap Fraud',
       'EWallet Related Fraud', 'Data Breach/Theft',
       'Cheating by Impersonation',
       'Denial of Service (DoS)/Distributed Denial of Service (DDOS) attacks',
       'FakeImpersonating Profile', 'Cryptocurrency Fraud',
       'Sexually Explicit Act', 'Sexually Obscene material',
       'Malware Attack', 'Business Email Compromise/Email Takeover',
       'Email Hacking', 'Hacking/Defacement',
       'Unauthorised Access/Data Breach', 'SQL Injection',
       'Provocative Speech for unlawful acts', 'Ransomware Attack',
       'Cyber Terrorism',
       'Child Pornography/Child Sexual Abuse Material (CSAM)',
       'Tampering with computer source documen

In [30]:
df['sub_category']=df['sub_category'].replace('DematDepository Fraud','Demat/Depository Fraud')
df['sub_category']=df['sub_category'].str.strip()
df['sub_category'].unique()

array(['Cyber Bullying/Stalking/Sexting', 'Fraud Call/Vishing',
       'Online Gambling  Betting', 'Online Job Fraud',
       'UPI Related Frauds', 'Internet Banking Related Fraud',
       'Rape/Gang Rape-Sexually Abusive Content', 'Other',
       'Profile Hacking Identity Theft',
       'Debit/Credit Card Fraud Or SIM Swap Fraud',
       'EWallet Related Fraud', 'Data Breach/Theft',
       'Cheating by Impersonation',
       'Denial of Service (DoS)/Distributed Denial of Service (DDOS) attacks',
       'FakeImpersonating Profile', 'Cryptocurrency Fraud',
       'Sexually Explicit Act', 'Sexually Obscene material',
       'Malware Attack', 'Business Email Compromise/Email Takeover',
       'Email Hacking', 'Hacking/Defacement',
       'Unauthorised Access/Data Breach', 'SQL Injection',
       'Provocative Speech for unlawful acts', 'Ransomware Attack',
       'Cyber Terrorism',
       'Child Pornography/Child Sexual Abuse Material (CSAM)',
       'Tampering with computer source documen

In [31]:
df['sub_category']=df['sub_category'].replace('Website DefacementHacking','Defacement/Hacking')
df['sub_category']=df['sub_category'].str.strip()
df['sub_category'].unique()

array(['Cyber Bullying/Stalking/Sexting', 'Fraud Call/Vishing',
       'Online Gambling  Betting', 'Online Job Fraud',
       'UPI Related Frauds', 'Internet Banking Related Fraud',
       'Rape/Gang Rape-Sexually Abusive Content', 'Other',
       'Profile Hacking Identity Theft',
       'Debit/Credit Card Fraud Or SIM Swap Fraud',
       'EWallet Related Fraud', 'Data Breach/Theft',
       'Cheating by Impersonation',
       'Denial of Service (DoS)/Distributed Denial of Service (DDOS) attacks',
       'FakeImpersonating Profile', 'Cryptocurrency Fraud',
       'Sexually Explicit Act', 'Sexually Obscene material',
       'Malware Attack', 'Business Email Compromise/Email Takeover',
       'Email Hacking', 'Hacking/Defacement',
       'Unauthorised Access/Data Breach', 'SQL Injection',
       'Provocative Speech for unlawful acts', 'Ransomware Attack',
       'Cyber Terrorism',
       'Child Pornography/Child Sexual Abuse Material (CSAM)',
       'Tampering with computer source documen

In [32]:
df['sub_category']=df['sub_category'].replace('Damage to computer computer systems etc','Damage to Computer Systems')
df['sub_category']=df['sub_category'].str.strip()
df['sub_category'].unique()

array(['Cyber Bullying/Stalking/Sexting', 'Fraud Call/Vishing',
       'Online Gambling  Betting', 'Online Job Fraud',
       'UPI Related Frauds', 'Internet Banking Related Fraud',
       'Rape/Gang Rape-Sexually Abusive Content', 'Other',
       'Profile Hacking Identity Theft',
       'Debit/Credit Card Fraud Or SIM Swap Fraud',
       'EWallet Related Fraud', 'Data Breach/Theft',
       'Cheating by Impersonation',
       'Denial of Service (DoS)/Distributed Denial of Service (DDOS) attacks',
       'FakeImpersonating Profile', 'Cryptocurrency Fraud',
       'Sexually Explicit Act', 'Sexually Obscene material',
       'Malware Attack', 'Business Email Compromise/Email Takeover',
       'Email Hacking', 'Hacking/Defacement',
       'Unauthorised Access/Data Breach', 'SQL Injection',
       'Provocative Speech for unlawful acts', 'Ransomware Attack',
       'Cyber Terrorism',
       'Child Pornography/Child Sexual Abuse Material (CSAM)',
       'Tampering with computer source documen

## Now we process the creminalinfo data

In [33]:
df.isnull().sum()

category              0
sub_category          0
crimeaditionalinfo    0
dtype: int64

## Now we lower the all the data it is good keep data in one case

In [34]:
df['crimeaditionalinfo']=df['crimeaditionalinfo'].str.lower()
df['category']=df['category'].str.lower()
df['sub_category']=df['sub_category'].str.lower()


In [35]:
df

category                     sub_category  \
0      online and social media related crime  cyber bullying/stalking/sexting   
1                     online financial fraud               fraud call/vishing   
2                   online gambling  betting         online gambling  betting   
3      online and social media related crime                 online job fraud   
4                     online financial fraud               fraud call/vishing   
...                                      ...                              ...   
93681                 online financial fraud   internet banking related fraud   
93682                 online financial fraud            ewallet related fraud   
93683                 online financial fraud               upi related frauds   
93684  online and social media related crime         online matrimonial fraud   
93685                 online financial fraud               fraud call/vishing   

                                      crimeaditionalinfo  
0      i had continue received random calls and abusi...  
1      the above fraudster is continuously messaging ...  
2      he is acting like a police and demanding for m...  
3      in apna job i have applied for job interview f...  
4      i received a call from lady stating that she w...  
...                                                  ...  
93681  identity theft   smishing sms fraud  creditdeb...  
93682  received call from  number asking about phone ...  
93683  cyber stalking   blackmailing   phonesmsvoip c...  
93684  call karke bola ki aapka lotary laga ha aru ac...  
93685  there is app name koko loan app they send the ...  

[93665 rows x 3 columns]

In [36]:
df.to_csv('../processed_data/processed_train.csv', index=False)


In [37]:
# Check class distribution in training set
print(df['sub_category'].value_counts())
print(df['category'].value_counts())

sub_category
upi related frauds                                                      26843
other                                                                   10877
debit/credit card fraud or sim swap fraud                               10802
internet banking related fraud                                           8871
fraud call/vishing                                                       5802
cyber bullying/stalking/sexting                                          4089
ewallet related fraud                                                    4047
rape/gang rape-sexually abusive content                                  2822
fakeimpersonating profile                                                2299
profile hacking identity theft                                           2072
cheating by impersonation                                                1987
sexually obscene material                                                1838
sexually explicit act                              